### PR TITLE
feat: Add visionOS as device family

### PR DESCRIPTION
**Pull request by brustolin**
Added `visionOS` as device family type. This will be used to filter events.

### CODE REVIEW

**Review by armcknight — COMMENTED** [review]
Following on from #3328. I think we'll also need to update a couple other things to use Xcode 15.2, some or all of these: 
- ci-select-xcode.sh to use Xcode 15.2 as a default: https://github.com/getsentry/sentry-cocoa/blob/94d8eb36ac3c56b44fecc9c4d13b41f9b24f34c2/scripts/ci-select-xcode.sh#L11
- the build GHA workflow to build and validate the xcframework: https://github.com/getsentry/sentry-cocoa/blob/94d8eb36ac3c56b44fecc9c4d13b41f9b24f34c2/.github/workflows/build.yml#L97
- the unit test GHA version matrix, 14.3 is the latest Xcode version we currently use (we should also update to run iOS 17 separately once we get xcode 15.2 support; see https://github.com/getsentry/sentry-cocoa/issues/3329): https://github.com/getsentry/sentry-cocoa/blob/94d8eb36ac3c56b44fecc9c4d13b41f9b24f34c2/.github/workflows/test.yml#L110-L114

**Comment by brustolin** [comment]
>Following on from https://github.com/getsentry/sentry-cocoa/pull/3328. I think we'll also need to update a couple other things to use Xcode 15.2, some or all of these:

Yes, but I think we should create a different PR for this things.

**Comment by brustolin** [comment]
@philipphofmann do you think we can go ahead with this?

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### Features
 
+- Add visionOS as device family (#3548)
```
Comment (philipphofmann) [inline CHANGELOG.md:7]: `m`: This needs to move up to the unreleased section

**Review by philipphofmann — APPROVED** [review]
Yes, sure. LGTM.

**Pull request merged**